src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -1,92 +1,113 @@
-import HTTPError from "../../models/HTTPError";
-import {ERRORS, RECORD_COMPLETENESS_ENUM, VEHICLE_TYPE} from "../../assets/Enums";
-import * as coreMandatoryValidation from "./CoreMandatoryValidations";
-import {ObjectSchema} from "@hapi/joi";
-import * as nonCoreMandatoryValidation from "./NonCoreMandatoryValidations";
-import { Vehicle, Trailer } from "../../../@Types/TechRecords";
-
-const validateRecordCompleteness = (validationSchema: ObjectSchema | undefined, techRecordFields: any) => {
-  if (validationSchema) {
-    return validationSchema.validate(techRecordFields, {stripUnknown: true});
-  } else {
-    return undefined;
+import { ValidationResult, ObjectSchema } from '@hapi/joi';
+import { Vehicle, Trailer } from '../../../@Types/TechRecords';
+import { ERRORS, RECORD_COMPLETENESS_ENUM, VEHICLE_TYPE } from '../../assets';
+import HTTPError from '../../models/HTTPError';
+import {
+  trlCoreMandatoryVehicleAttributes,
+  psvHgvCarLgvMotoCoreMandatoryVehicleAttributes,
+  carCoreMandatorySchema,
+  hgvCoreMandatorySchema,
+  lgvCoreMandatorySchema,
+  motorcycleCoreMandatorySchema,
+  psvCoreMandatorySchema,
+  trlCoreMandatorySchema
+} from './CoreMandatoryValidations';
+import { psvNonCoreMandatorySchema, hgvNonCoreMandatorySchema, trlNonCoreMandatorySchema } from './NonCoreMandatoryValidations';
+
+export function computeRecordCompleteness(vehicle: Vehicle): string {
+  if (!vehicle.systemNumber) {
+    throw new HTTPError(400, ERRORS.SYSTEM_NUMBER_GENERATION_FAILED);
   }
-};
 
-const validateVehicleAttributes = (vehicleType: string, vehicleAttributes: any) => {
-  if (vehicleType !== VEHICLE_TYPE.TRL) {
-    return validateRecordCompleteness(coreMandatoryValidation.psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, vehicleAttributes);
+  const generalErrors = validateVehicleAttributes(vehicle)?.error;
+
+  if (generalErrors) {
+    console.log("general errors: ",generalErrors);
+    return RECORD_COMPLETENESS_ENUM.SKELETON;
+  }
+
+  const mandatoryAttributesValidationResult = validateMandatoryTechRecordAttributes(vehicle);
+
+  const isCoreMandatoryValid = !mandatoryAttributesValidationResult.coreValidationResult?.error;
+
+  const isNonCoreMandatoryValid = !mandatoryAttributesValidationResult.nonCoreValidationResult?.error;
+
+  if (isCoreMandatoryValid && isNonCoreMandatoryValid) {
+    return RECORD_COMPLETENESS_ENUM.COMPLETE;
+  } else if (isCoreMandatoryValid) {
+    return RECORD_COMPLETENESS_ENUM.TESTABLE;
   } else {
```

Comment:
The 2nd check on isCoreMandatoryValid seemed a bit wasteful

```suggestion
  if (isCoreMandatoryValid) {
    if (isNonCoreMandatoryValid) {
       return RECORD_COMPLETENESS_ENUM.COMPLETE;
    }
    return RECORD_COMPLETENESS_ENUM.TESTABLE;
  }
  return RECORD_COMPLETENESS_ENUM.SKELETON
```

Reply:
Because the actual check is already made on line 31 and the result stored in this variable, the whole check does not run again, it only does `booleanValue === true`.

Thus, it omits the nesting and conforms with the `if, else if, else` standard - apply the strictest check at first and start loosening with each subsequent check.